neptune/new/internal/operation_processors/async_operation_processor.py

```diff
@@ -89,24 +90,59 @@ def flush(self):
     def start(self):
         self._consumer.start()
 
+    def _wait_for_queue_empty(self, initial_queue_size: int, seconds: Optional[float]):
+        waiting_start = monotonic()
+        max_wait_time = self.STOP_QUEUE_MAX_TIME_NO_CONNECTION_SECONDS if seconds is None else seconds
+        if initial_queue_size > 0:
+            if self._consumer.last_backoff_time > 0:
+                click.echo(f"We have been experiencing connection interruptions during your run. "
+                           f"Client will now try to resume connection and sync data for the next "
+                           f"{max_wait_time} seconds. "
+                           f"You can also synchronize manually using neptune sync command.",
+                           sys.stderr)
+            click.echo(f"Waiting for the remaining {initial_queue_size} operations to synchronize with Neptune. "
+                       f"Do not kill you application.",
+                       sys.stderr)
+        if seconds is not None:
+            return self._queue.wait_for_empty(seconds)
+        else:
+            size_remaining = initial_queue_size
+            while size_remaining > 0:
+                self._queue.wait_for_empty(self.STOP_QUEUE_STATUS_UPDATE_FREQ_SECONDS)
+                size_remaining = self._queue.size()
+                already_synced = initial_queue_size - size_remaining
+                already_synced_proc = (already_synced / initial_queue_size) * 100
+                if size_remaining == 0:
+                    click.echo(f"All {initial_queue_size} operation synced, thanks for waiting!")
```

Comment:
operation -> operations

Shouldn't we have return statement here?

Reply:
we're in `while size_remaining > 0` loop, so it's not really necessary -- but I'll try and make the code cleaner